notebooks/concurrent_tasks.ipynb

```diff
@@ -0,0 +1,1226 @@
+{
```

Comment:
<p>We wrap multiple generators into a single component that internally will run them concurrently in different threads. To simplify the threads orchestration, we use the <code>to_thread</code> function from <code>asyncio</code>. The component itself will stay synchronous, exposing the usual <code>run</code> method.    </p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/deepset-ai/haystack-cookbook/pull/81/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='11'/>

Reply:
<p>done</p>